code/modules/reagents/chemistry/items.dm

```diff
@@ -162,3 +131,212 @@
+		attack_verb_simple = string_list(list("burn", "sing"))
+		START_PROCESSING(SSobj, src)
+	else
+		hitsound = "swing_hit"
+		force = 0
+		attack_verb_continuous = null //human_defense.dm takes care of it
+		attack_verb_simple = null
+		STOP_PROCESSING(SSobj, src)
+	set_light_on(lit)
+	update_icon()
+
+/obj/item/burner/extinguish()
+	set_lit(FALSE)
+
+/obj/item/burner/attack_self(mob/living/user)
+	if(user.is_holding(src))
+		if(lit)
+			set_lit(FALSE)
+			user.visible_message("<span class='notice'>[user] snuffs out [src]'s flame.</span>")
+	else
+		. = ..()
+
+/obj/item/burner/attack(mob/living/carbon/M, mob/living/carbon/user)
+	if(lit && M.IgniteMob())
+		message_admins("[ADMIN_LOOKUPFLW(user)] set [key_name_admin(M)] on fire with [src] at [AREACOORD(user)]")
+		log_game("[key_name(user)] set [key_name(M)] on fire with [src] at [AREACOORD(user)]")
+	return ..()
+
+/obj/item/burner/process()
+	var/current_heat = 0
+	var/number_of_burning_reagents = 0
+	for(var/_reagent in reagents.reagent_list)
+		var/datum/reagent/reagent = _reagent
+		if(istype(reagent, /datum/reagent/consumable/ethanol))
+			current_heat += 2193//ethanol burns at 1970C (at it's peak)
+			number_of_burning_reagents += 1
+			reagents.remove_reagent(/datum/reagent/consumable/ethanol, 0.05)
+			continue
+
+		if(ispath(reagent, /datum/reagent/fuel))
+			current_heat += 1725//Refined slightly
+			number_of_burning_reagents += 1
+			reagents.remove_reagent(/datum/reagent/fuel, 0.1)
+			continue
+
+		if(istype(reagent, /datum/reagent/fuel/oil))
+			current_heat += 1200//Oil is crude
+			number_of_burning_reagents += 1
+			reagents.remove_reagent(/datum/reagent/fuel/oil, 0.025)//But lasts longer
+			continue
+
+		if(istype(reagent, /datum/reagent/toxin/plasma))//For fun
+			current_heat += 4500//plasma is hot!!
+			number_of_burning_reagents += 1
+			reagents.remove_reagent(/datum/reagent/toxin/plasma, 0.15)//But burns fast
+			continue
+
+	if(!number_of_burning_reagents)
+		set_lit(FALSE)
+		heat = 0
+		return
+	open_flame()
+	current_heat /= number_of_burning_reagents
+	heat = current_heat
+
+/obj/item/burner/get_temperature()
+	return lit * heat
+
+/obj/item/burner/oil
+	name = "Oil burner"
+	reagent_type = /datum/reagent/fuel/oil
+	grind_results = list(/datum/reagent/fuel/oil = 5, /datum/reagent/silicon = 10)
+
+/obj/item/burner/fuel
+	name = "Fuel burner"
+	reagent_type = /datum/reagent/fuel
+	grind_results = list(/datum/reagent/fuel = 5, /datum/reagent/silicon = 10)
+
+/obj/item/thermometer
+	name = "thermometer"
+	desc = "A thermometer for checking a beaker's temperature"
+	icon_state = "thermometer"
+	icon = 'icons/obj/chemical.dmi'
+	item_flags = NOBLUDGEON
+	w_class = WEIGHT_CLASS_TINY
+	grind_results = list(/datum/reagent/mercury = 5)
+	var/obj/item/reagent_containers/attached_beaker
+
+/obj/item/thermometer/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(istype(target, /obj/item/reagent_containers))
```

Comment:
Given how many of these checks are, you could make a define for them, similar to `isitem()` and `isstack()`

Reply:
Sure, should I change all instances?

Reply:
I did so and replaced my use cases